llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6631,6 +6630,30 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
 
   BranchInst *RangeCheckBranch = nullptr;
 
+  // Grow the table to cover all possible index values to avoid the range check.
+  // It will use the default result to fill in the table hole later, so make
+  // sure it exist.
+  if (UseSwitchConditionAsTableIndex && HasDefaultResults) {
+    ConstantRange CR = computeConstantRange(TableIndex, /* ForSigned */ false);
+    // Grow the table shouldn't have any size impact by checking
+    // WouldFitInRegister.
+    // TODO: Consider growing the table also when it doesn't fit in a register
+    // if no optsize is specified.
+    // Check that the constant range hasn't wrapped.
+    const APInt &UpperBound = CR.getUpper();
+    if (UpperBound.getZExtValue() > CR.getLower().getZExtValue() &&
+        all_of(ResultTypes, [&](const auto &KV) {
+          return SwitchLookupTable::WouldFitInRegister(
+              DL, UpperBound.getLimitedValue(), KV.second /* ResultType */);
```

Comment:
Why `getLimitedValue()` here but `getZextValue()` everywhere else?

Reply:
Oh, thanks, I'll unify the update to `getLimitedValue()`